### PR TITLE
Move Logic Into The Components That Actually Use Them

### PR DESCRIPTION
**Pull request by AdrianAndersen**
image

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -237,31 +189,25 @@ const Index: NextPage<{
                         onSettingsOpen={() => setIsSettingsOpen(true)}
                         onAgendaOpen={() => setIsAgendaOpen(true)}
                     />
-                    <WeekNavigator
```
Comment (mathiazom) [inline pages/index.tsx:240]: The move has altered the flex behavior. The schedule component should expand to fill the whole height.

Screenshot from 2023-07-30 08-45-24

Comment (AdrianAndersen) [reply pages/index.tsx:240]: Oh yeah, I did not forsee that. Luckily, I performed some funny circle composition, where WeekNavigator now is back in index.tsx. So I fixed this by moving the </Box> to where it originally was in https://github.com/mathiazom/sit-rezervo-confgen/pull/29/commits/90065a45e5f37a615f1eeee385a22f65db64e9d3 ✅

File: components/WeekNavigator.tsx [hunk 2]
```diff
@@ -1,30 +1,74 @@
...
+        const nextSchedule = await fetch("api/schedule", {
+            method: "POST",
+            body: JSON.stringify({ weekOffset: nextWeekOffset }),
+        }).then((r) => r.json());
+        if (nextSchedule != undefined) {
+            setCachedSchedules({ ...cachedSchedules, [nextWeekOffset]: nextSchedule });
+        }
+    }
+
+    const weekNumber = DateTime.fromISO(initialCachedSchedules[weekOffset]!.days[0]!.date).weekNumber;
```
Comment (mathiazom) [inline components/WeekNavigator.tsx]: This occurs when navigating two weeks back

image

Comment (mathiazom) [reply components/WeekNavigator.tsx]: Introduced in 3f631f259699cd9edc392bc2036090e353f3b074

Comment (AdrianAndersen) [reply components/WeekNavigator.tsx]: Fixed in https://github.com/mathiazom/sit-rezervo-confgen/pull/29/commits/e89794db6f03dd898c48354bbfb03c9c705f6fb2

Let me know if you want me to rebase and amend this fix to the original commit (https://github.com/mathiazom/sit-rezervo-confgen/commit/3f631f259699cd9edc392bc2036090e353f3b074)

**Review by mathiazom — APPROVED** [review]
🚀

**Pull request merged**